llama-index-core/llama_index/core/memory/vector_memory.py

```diff
@@ -144,58 +147,37 @@ def get_all(self) -> List[ChatMessage]:
 
     def _commit_node(self, override_last: bool = False) -> None:
         """Commit new node to vector store."""
-        # commit to vector store
-        node_id = str(uuid.uuid4())
-        # create subnodes for each message
-        sub_dicts = []
-        for msg in self.chat_store.get_messages(self.cur_user_msg_key):
-            sub_dicts.append(_stringify_chat_message(msg))
-
-        if not sub_dicts:
+        if self.cur_batch_textnode.text == "":
             return
 
-        # now create a "super" node that contains all subnodes as metadata
-        # this metadata is excluded from embedding and LLM synthesis
-        # the concatenated text is put into the super node text field
-        super_node = TextNode(
-            text=" ".join(
-                [str(sub_dicts[i]["content"]) for i in range(len(sub_dicts))]
-            ),
-            id_=node_id,
-            metadata={"sub_dicts": sub_dicts},
-            excluded_embed_metadata_keys=["sub_dicts"],
-            excluded_llm_metadata_keys=["sub_dicts"],
-        )
-
         if override_last:
             # delete the last node
             # This is needed since we're updating the last node in the vector
             # index as its being updated. When a new user-message batch starts
             # we already will have the last user message group committed to the
             # vector store index and so we don't need to override_last (i.e. see
             # logic in self.put().)
-            last_node_id = self.chat_store.delete_last_message(
-                self.chat_store_key
-            ).content
-            self.vector_index.delete_nodes([last_node_id])
+            self.vector_index.delete_nodes([self.cur_batch_textnode.id_])
```

Comment:
I guess I just realized this memory is limited to vector stores that implement this 😅 I wonder if we should note this somewhere, which vector stores support this at the time of writing

Reply:
Good point! Where do you think it would make most sense to add this note? In the class string?

Also, do you know off chance which vector stores support this at this time (I presume this also applies for `.clear()`)?

Reply:
<img width="588" alt="image" src="https://github.com/run-llama/llama_index/assets/92402603/fb8a0c06-c684-4fda-b3d7-95597226006f">

good ol' string search. I know `SimpleVectorStore` supports it as well.